server/admin-api/services/lifecycle.ts

```diff
@@ -123,10 +158,20 @@ const subscribeLifecycleMethods = async (modelName) => {
 
       // Delete the path entity.
       async beforeDeleteMany(event) {
-        const ids = event.params.where.$and[0].id.$in;
-        for (let i = 0; i < ids.length; i++) {
-          // eslint-disable-next-line no-await-in-loop
-          await deleteEntity(event, ids[i]);
+        const result = await strapi.db.query(modelName).findMany({
```

Comment:
Is there a specific reason you didn't use the entityService layer here?

Reply:
The params of the event directly map to using the strapi query. The entity service does not use `where` anywhere, so they would be incompatible. Doing it this way we can be certain that it would fetch the same entries as are being deleted :)